python/taichi/lang/misc.py

```diff
@@ -442,6 +442,15 @@ def init(arch=None,
     print(f'[Taichi] Starting on arch={_ti_core.arch_name(cfg.arch)}')
 
     # user selected visible device
+    if os.environ.get("CUDA_VISIBLE_DEVICES"
+                      ) and not os.environ.get("TI_VISIBLE_DEVICE"):
+        os.environ["TI_VISIBLE_DEVICE"] = os.environ.get(
+            "CUDA_VISIBLE_DEVICES")
+    elif os.environ.get("TI_VISIBLE_DEVICE"
+                        ) and not os.environ.get("CUDA_VISIBLE_DEVICES"):
+        os.environ["CUDA_VISIBLE_DEVICES"] = os.environ.get(
```

Comment:
Not sure it works this way ( override CUDA_VISIBLE_DEVICES in Python's os.environ() ). We probably need some test cases to verify/guard this change, but before that we need a CI bot with more than one Nvidia cards...